core/blockchain_reader.go

```diff
@@ -439,6 +439,21 @@ func (bc *BlockChain) GetBlockStatusByNum(number uint64) (uint8, common.Hash) {
 	status, hash := rawdb.ReadBlockStatusByNum(bc.db, new(big.Int).SetUint64(number))
 	// Cache the found status for next time and return
 	// Only deterministic data is saved, and data tracking is required only at the beginning of startup
+
+	if status != types.BasFinalized {
+		lastFinalizedBlockNumber := bc.GetLastFinalizedBlockNumber()
+		if number < lastFinalizedBlockNumber {
+			currentBlock := bc.GetBlockByNumber(number)
+			lastFinalizedBlock := bc.GetBlockByNumber(lastFinalizedBlockNumber)
+			bFiliation, err := bc.IsFiliation(&types.RangeEdge{Hash: currentBlock.Hash(), Number: currentBlock.Number()},
```

Comment:
`GetBlockByNumber` will only get block from the canonical chain, so the `currentBlock` and the `lastFinalizedBlock` here will definitely in the same branch.